### ISSUE TITLE
Span Json Format Different causing issues

### ISSUE THREAD

**Issue by prateekvjain** [openzipkin/zipkin#1126]
Hi All,
Spring sleuth have given the kafka binder option and the Span data format generated via the format that is consumed directly by Zipkin Query server.

Currently I have writtent a Kafka Producer Api, which based on the topic name would publish the span data and then will have it consumed by logstash and then written to zipkin query.

Can we have only one JSON format produced by all the span producing API.

input {
  kafka {
    zk_connect => "***_.us-west-2.compute.amazonaws.com:2181"
    topic_id => "???_queue"
  }
}
output {
  http {
    url => "http://**_*west-2.compute.amazonaws.com:9411/api/v1/spans"
    http_method => "post"
  }
}

**Comment by codefromthecrypt** [comment]
TL;DR I feel your pain, as I've mentioned this myself a couple times. However, the route isn't to ask the entire zipkin ecosystem to change to adopt this format. The following issue is tracking this https://github.com/spring-cloud/spring-cloud-sleuth/issues/253

There are good reasons for spring-cloud-stream to envelope data, because it is trying to do things portable across different messaging transports. This isn't a good enough reason to muddy zipkin's data format.

For example, this data has nothing to do with the payload, intentionally! There's also no cross-language tool that shares this format. In other words, we'd be special-casing spring-cloud-stream, but impacting several languages who don't have a tool that reads this, and probably couldn't be runtime portable with spring-cloud-stream. In other words, this is the wrong way to look at the problem.

I'd explore some way of making sleuth/stream send zipkin formatted messages to a dedicated zipkin topic. For example, sending in a way that doesn't require affecting the payload. If that isn't possible, another option would be to add a KafkaZipkinSpanReporter, referencing the one in brave.

https://github.com/openzipkin/brave/blob/master/brave-spancollector-kafka/src/main/java/com/github/kristofa/brave/kafka/KafkaSpanCollector.java

Kafka reporters in any language are <100 lines of code, and it is normal case in zipkin to expect transports such as http, scribe and kafka to not share the actual transport code or metadata. What does work is sharing a data format, so we should focus on how to achieve that.
